File: pc_builder_service/internal/domain/entities/cpu.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type CPU struct {
	gorm.Model
	
	ID 					uuid.UUID 		`gorm:"type:uuid primaryKey"`
	Gen 				string 			`gorm:"type:varchar(10) not null"`
	Family 				string 			`gorm:"type:varchar(20) not null"`
	Series 				string 			`gorm:"type:varchar(20) not null"`
	Cores 				int32 			`gorm:"type:integer not null"`
	Threads 			int32 			`gorm:"type:integer not null"`
	BaseClock 			float32 		`gorm:"type:real not null"`
	MaxClock 			float32 		`gorm:"type:real not null"`
	Cache 				int32 			`gorm:"type:integer not null"`
	Socket 				string 			`gorm:"type:varchar(20) not null"`
	Graphics 			bool 			`gorm:"type:boolean not null"`
	OC 					bool 			`gorm:"type:boolean not null"`
	RecommendedPower 	int32 			`gorm:"type:integer not null"`
	ReleaseDate 		time.Time 		`gorm:"type:timestamp not null"`
	AvgPrice 			float32 		`gorm:"type:real not null"`
	Img 				[]byte 			`gorm:"type:bytea"`
	CreatedAt 			time.Time 		`gorm:"type:timestamp not null"`
	UpdatedAt 			time.Time 		`gorm:"type:timestamp"`
}

File: benchmark_service/src/main/java/com/buildpc/benchmark_service/entities/CPU.java
package com.buildpc.benchmark_service.entities;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.sql.Date;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "cpus")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CPU {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private UUID id;
	private String brand;
	private String gen;
	private String family;
	private String series;
	private Integer cores;
	private Integer threads;
	private Float baseClock;
	private Float maxClock;
	private Integer cache;
	private String socket;
	private Boolean graphics;
	private Boolean oc;
	private Integer recommendedPower;
	private LocalDateTime releaseDate;
	private Float avgPrice;
	private byte[] img;

	@CreatedDate
	private LocalDateTime createdAt;

	@LastModifiedDate
	private LocalDateTime updatedAt;
}
